packages/helpers/scripts/ast_grep.js

```diff
@@ -86,27 +86,14 @@ export function ast_grep() {
 
                     source.prependLeft(range.start.index, `exports._ = exports.${func_name} = `);
```

Comment:
```suggestion
                    source.prependLeft(range.start.index, `exports._ = `);
```

There is also a similar statement on line 73.

Reply:
I think we can just remove the whole forEach part:

```js
match
  .findAll({
      rule: {
          pattern: func_name,
          kind: "identifier",
          inside: { kind: "assignment_expression", field: "left" },
      },
  })
  .forEach((match) => {
      const range = match.range();
  
      source.prependLeft(range.start.index, `exports._ = exports.${func_name} = `);
  });
```

This is doing the same thing:

```js
source.update(
  export_start,
  export_end,
  `exports._ = exports.${func_name} = ${func_name};`,
);
```

Reply:
I think it can also be just:

```js
`exports._ = ${func_name};`
```

Since we are going to access `_` only?

Reply:
Function declarations are hoisted. But using function expressions directly might cause issues. This is especially true in cases of self-recursion within the function. You need to verify it.

Reply:
> I think we can just remove the whole forEach part:
> 
> ```js
> match
>   .findAll({
>       rule: {
>           pattern: func_name,
>           kind: "identifier",
>           inside: { kind: "assignment_expression", field: "left" },
>       },
>   })
>   .forEach((match) => {
>       const range = match.range();
>   
>       source.prependLeft(range.start.index, `exports._ = exports.${func_name} = `);
>   });
> ```
> 
> This is doing the same thing:
> 
> ```js
> source.update(
>   export_start,
>   export_end,
>   `exports._ = exports.${func_name} = ${func_name};`,
> );
> ```

This is used to modify the internal references within a function. Consider the following example.

https://github.com/swc-project/swc/blob/1ab406cd7aa19ea333a8462b0cd496ceb3e39ac1/packages/helpers/esm/_construct.js

```JavaScript
import { _is_native_reflect_construct } from "./_is_native_reflect_construct.js";
import { _set_prototype_of } from "./_set_prototype_of.js";
export function _construct(Parent, args, Class) {
    if (_is_native_reflect_construct()) _construct = Reflect.construct;
    else {
        _construct = function construct(Parent, args, Class) {
            var a = [null];
            a.push.apply(a, args);
            var Constructor = Function.bind.apply(Parent, a);
            var instance = new Constructor();

            if (Class) _set_prototype_of(instance, Class.prototype);

            return instance;
        };
    }

    return _construct.apply(null, arguments);
}
export { _construct as _ };
```

Please note that the exported `_construct` will be updated within the function body.

Reply:
Cool, had the same fix lol. :)

added comment why we find from `tree.root`!